packages/trie/src/proof/index.ts

```diff
@@ -64,4 +66,124 @@ export function verifyTrieRangeProof(
+  const opStack = proof.map((nodeValue) => {
+    let key = Uint8Array.from(trie['hash'](nodeValue))
+    key = trie['_opts'].keyPrefix ? concatBytes(trie['_opts'].keyPrefix, key) : key
+    return {
+      type: 'put',
+      key,
+      value: nodeValue,
+    } as PutBatch
+  })
+
+  if (shouldVerifyRoot) {
+    if (opStack[0] !== undefined && opStack[0] !== null) {
+      if (!equalsBytes(trie.root(), opStack[0].key)) {
+        throw new Error('The provided proof does not have the expected trie root')
+      }
+    }
+  }
+
+  await trie['_db'].batch(opStack)
+  if (opStack[0] !== undefined) {
+    return opStack[0].key
+  }
+}
+
+/**
+ * Verifies a proof by putting all of its nodes into a trie and attempting to get the proven key. An (EIP-1186)[https://eips.ethereum.org/EIPS/eip-1186] proof
+ * contains the encoded trie nodes from the root node to the leaf node storing state data. A static version of this function exists with the same name.
+ * @param rootHash Root hash of the trie that this proof was created from and is being verified for
+ * @param key Key that is being verified and that the proof is created for
+ * @param proof an EIP-1186 proof to verify the key against
+ * @throws If proof is found to be invalid.
+ * @returns The value from the key, or null if valid proof of non-existence.
+ */
+export async function verifyProof(
+  trie: Trie,
+  rootHash: Uint8Array,
+  key: Uint8Array,
+  proof: Proof,
+): Promise<Uint8Array | null> {
+  trie['DEBUG'] &&
+    trie['debug'](
+      `Verifying Proof:\n|| Key: ${bytesToHex(key)}\n|| Root: ${bytesToHex(
+        rootHash,
+      )}\n|| Proof: (${proof.length}) nodes
+  `,
+      ['VERIFY_PROOF'],
+    )
+  const proofTrie = new Trie({
+    root: rootHash,
+    useKeyHashingFunction: trie['_opts'].useKeyHashingFunction,
+    common: trie['_opts'].common,
+  })
+  try {
+    await updateFromProof(proofTrie, proof, true)
+  } catch (e: any) {
+    throw new Error('Invalid proof nodes given')
+  }
+  try {
+    trie['DEBUG'] &&
+      trie['debug'](`Verifying proof by retrieving key: ${bytesToHex(key)} from proof trie`, [
+        'VERIFY_PROOF',
+      ])
+    const value = await proofTrie.get(trie['appliedKey'](key), true)
+    trie['DEBUG'] && trie['debug'](`PROOF VERIFIED`, ['VERIFY_PROOF'])
+    return value
+  } catch (err: any) {
+    if (err.message === 'Missing node in DB') {
+      throw new Error('Invalid proof provided')
+    } else {
+      throw err
+    }
+  }
+}
+
+/**
+ * Create a trie from a given (EIP-1186)[https://eips.ethereum.org/EIPS/eip-1186] proof. An EIP-1186 proof contains the encoded trie nodes from the root
+ * node to the leaf node storing state data. This function does not check if the proof has the same expected root. A static version of this function exists
+ * with the same name.
+ * @param proof an EIP-1186 proof to update the trie from
+ * @deprecated Use `updateFromProof`
+ */
+export async function fromProof(trie: Trie, proof: Proof): Promise<void> {
```

Comment:
This function feels duplicative of `createTrieFromProof` in the `constructors.ts`.  Can you think of a reason we'd need it here as well?

Reply:
I'd agree